File: proyectoTaeKwonDo/src/app/_services/examen.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Examen } from '../_models/examen';

@Injectable({
  providedIn: 'root'
})
export class ExamenService {

  API_URI = 'http://localhost:8080';

  constructor(private http: HttpClient) { }

  getExamenes(){
    return this.http.get(this.API_URI+'/examen');
  }

  getExamen(id_examen: number){
    return this.http.get(this.API_URI+'/examen/'+id_examen);
  }

  createExamen(examen: Examen){
    return this.http.post(this.API_URI+'/examen', examen);
  }

  updateExamen(examen: Examen){
    return this.http.put(this.API_URI+'/examen/'+examen.id_examen, examen);
  }

  deleteExamen(id_examen: number){
    return this.http.delete(this.API_URI+'/examen/'+id_examen);
  }
}

File: taekwondo Spring/taekwondo/src/main/java/com/taekwondo/service/ExamenService.java
package com.taekwondo.service;

import java.util.List;

import org.springframework.http.ResponseEntity;

import com.taekwondo.model.Examen;

public interface ExamenService {
	public abstract List<Examen> getExamenes();
	public abstract ResponseEntity<Object> getExamen(int id);
	public abstract ResponseEntity<Object> createExamen(Examen e);
	public abstract void updateExamen(int id, Examen e);
	public abstract void deleteExamen(int id);
}
